jams/core.py

```diff
@@ -496,49 +498,21 @@ class Sandbox(JObject):
     pass
 
 
-class JamsFrame(pd.DataFrame):
-    '''A data-frame class for JAMS.
+Observation = namedtuple('Observation',
+                         ['time', 'duration', 'value', 'confidence'])
+'''Core observation type: (time, duration, value, confidence).'''
 
-    This automates certain niceties, such as timestamp
-    conversion and serialization.
-    '''
-
-    __dense = False
-
-    def __init__(self, data=None, index=None, columns=None, dtype=None):
-        '''Construct a new JamsFrame object.
-
-        Parameters
-        ----------
-        data
-            Optional data for the new JamsFrame, in any format supported
-            by `pandas.DataFrame.__init__`.
 
-            Fields must be `['time', 'duration', 'value', 'confidence']`.
+class AnnotationData(object):
```

Comment:
> if we're not going to inherit from it, I'd propose making obs some level of private (dunders?) so that users are discouraged from using it as part of the API.

Yeah, I'd be down with that.  I'd say sunder not dunder, ie `data._obs`.



> Also worth mentioning here: What gets added to an AnnotationData object? Observations or the fields of an observation? or, said differently, who is responsible for creating the Observation object, the user or the AnnotationData object? because, if it's the former, I think that's an argument for inheritance.

It depends: if you're adding a bunch of records at once, it's typically in deserialized form (ie after a json parse), so it'd be list-of-dict/dict-of-list in.

The core add routine takes the fields for a single observation record and packs them in an `Observation` for adding.

I don't see what this has to do with inheritance though?



> related pop quiz: in Jams, who creates an Annotation when adding it to a JAMS object, the user or the owning object? (because we should follow that paradigm for consistency)

Should we?  I see the `AnnotationData` class as relatively private, because it diverges somewhat from the schema. `Annotation` provides an interface for adding observations, and that's all a user should need.